src/main/java/org/opensearch/security/configuration/ConfigurationRepository.java

```diff
@@ -312,6 +358,75 @@ private void waitForSecurityIndexToBeAtLeastYellow() {
         }
     }
 
+    // viable for testing
+    protected Future<Void> initDefaultConfiguration() {
+        return threadPool.generic().submit(() -> {
+            if (!clusterService.state().metadata().hasIndex(securityIndex)) {
+                securityIndexHandler.createIndex(
+                    ActionListener.wrap(r -> uploadSecurityConfiguration(), e -> LOGGER.error("Couldn't create index {}", securityIndex, e))
+                );
+            } else {
+                // in case index was created and cluster stat has not been changed (e.g. restart if the node or something)
+                // just upload default configuration
+                uploadSecurityConfiguration();
+            }
+            return null;
+        });
+    }
+
+    private void uploadSecurityConfiguration() {
+        securityIndexHandler.uploadSecurityConfiguration(
+            resolveConfigDir(),
+            ActionListener.wrap(
+                configuration -> clusterService.submitStateUpdateTask(
+                    "init-security-configuration",
+                    new ClusterStateUpdateTask(Priority.IMMEDIATE) {
+                        @Override
+                        public ClusterState execute(ClusterState clusterState) throws Exception {
+                            return ClusterState.builder(clusterState)
+                                .putCustom(SecurityMetadata.TYPE, new SecurityMetadata(Instant.now(), configuration))
+                                .build();
+                        }
+
+                        @Override
+                        public void onFailure(String s, Exception e) {
+                            LOGGER.error(s, e);
+                        }
+                    }
+                ),
+                e -> LOGGER.error("Couldn't upload default configuration", e)
+            )
+        );
+    }
+
+    // visible for testing
+    protected Future<Void> reloadConfiguration(final SecurityMetadata securityMetadata) {
+        if (!initalizeConfigTask.isDone() && !configReloadInProcess.get()) {
+            configReloadInProcess.getAndSet(true);
```

Comment:
This isn't thread safe, see https://stackoverflow.com/questions/4501223/when-do-i-need-to-use-atomicboolean-in-java

Reply:
Compare & set should happen in the same function call, e.g

```
if (configReloadInProcess.compareAndSet(false, true)) {
    // initialize ...
}
```